### ISSUE TITLE
[Bug] APNet inference returns NaN probability for peptides containing specific residues (P/G/A/R) due to missing AAindex values

### ISSUE THREAD

**Issue by Adawi1201** [Wonka1999/AggNet#5]
### 问题描述

使用官方 `APNet.ckpt` 在 README 示例的 Hex142 数据集上推理时，有 2 条序列的预测概率为 `NaN`：

- `IIPFEQ` → NaN
- `NPQGGR` → NaN

其余 140 条序列正常。该问题同样影响 `train_apnet.py` 类自训练流程中的验证集（`roc_auc_score` 因 NaN 报错）。

### 复现步骤

```bash
python ./script/predict_amyloid.py \
  --fasta ./data/AmyHex/Hex142.fasta \
  --batch_size 256 \
  --checkpoint ./checkpoint/APNet.ckpt \
  --output ./APNet_results.csv
```

输出（节选）：

```
    peptide  probability        label
137  STVDIE     0.145792  non-amyloid
138  YQLENY     0.425879  non-amyloid
139  IIPFEQ          NaN  non-amyloid
140  FQKQQK     0.074112  non-amyloid
141  NPQGGR          NaN  non-amyloid
```

### 根因分析

定位到 `utils/dataset/Tokenizer.py` 中 `AAIndex.get_aaindex_table` 构建的 AAindex 表存在**源数据库缺失值**：`aaindex1` 中有 13 条>记录对个别氨基酸无定义值（值为 NaN），包括：

| 记录 | 缺失氨基酸 |
|---|---|
| AVBF000101 ~ AVBF000109（9 条） | P |
| YANJ020101 | A, G |
| GUYH850103 | G |
| ROSM880104 | P, R |
| ROSM880105 | P |

当序列包含上述氨基酸、且对应列被 `checkpoint/selected_aaindex.pt`（600 维）选中时，`data_module.py` 的 `collate_fn` 中 `batch_aaindex` 即含 NaN；NaN 经 `feature_mlp` 传播，最终 `cls_head` logits 为 NaN，softmax 后概率为 NaN。

实测：
- `IIPFEQ`（含 P/F）命中 7 个 NaN 列（索引 1604, 1605, 1608, 1609, 1611, 1655, 1656）
- `NPQGGR`（含 G）命中 1 个 NaN 列（索引 2784）

### 建议修复

在 `model/APNet/data_module.py` 的 `collate_fn` 中，对 AAindex 特征做 NaN 填充：

```python
batch_aaindex = self.tokenizer.aaindex.batch_encode(batch_seqs)[:, self.selected_aaindex]
batch_aaindex = torch.from_numpy(batch_aaindex).float()
batch_aaindex = torch.nan_to_num(batch_aaindex, nan=0.0)   # 新增
```

修复后 `IIPFEQ` → 0.0028、`NPQGGR` → 0.0001（均为 non-amyloid），Hex142 全量指标恢复：ACC 0.852 / AUC 0.909（论文 Table 1：0.873 / 0.913）。

官方的训练流程是否同样存在该 NaN？（若存在，则训练/推理行为一致，填充 0 是合理的；若不存在，请告知训练时对缺失 AAindex 值>的处理方式，以便对齐）。

---

### Description

When running inference with the official `APNet.ckpt` on the Hex142 dataset (README example), 2 sequences get `NaN` prediction probabilities:

- `IIPFEQ` → NaN
- `NPQGGR` → NaN

The other 140 sequences are fine. The same issue also breaks validation in training pipelines (e.g., `roc_auc_score` fails on NaN), which blocks reproduction of the paper.

### Reproduction

```bash
python ./script/predict_amyloid.py \
  --fasta ./data/AmyHex/Hex142.fasta \
  --batch_size 256 \
  --checkpoint ./checkpoint/APNet.ckpt \
  --output ./APNet_results.csv
```

Output (excerpt):

```
    peptide  probability        label
137  STVDIE     0.145792  non-amyloid
138  YQLENY     0.425879  non-amyloid
139  IIPFEQ          NaN  non-amyloid
140  FQKQQK     0.074112  non-amyloid
141  NPQGGR          NaN  non-amyloid
```

### Root Cause

The AAindex table built in `utils/dataset/Tokenizer.py` (`AAIndex.get_aaindex_table`) contains **NaN values inherited from the aaindex source database**: 13 records lack values for specific amino acids:

| Record | Missing residues |
|---|---|
| AVBF000101 ~ AVBF000109 (9 records) | P |
| YANJ020101 | A, G |
| GUYH850103 | G |
| ROSM880104 | P, R |
| ROSM880105 | P |

When a peptide contains those residues and the corresponding columns are selected by `checkpoint/selected_aaindex.pt` (600-dim), the `batch_aaindex` tensor in `collate_fn` (`model/APNet/data_module.py`) contains NaN. The NaN propagates through `feature_mlp` and produces NaN logits from `cls_head`, hence NaN probability after softmax.

Measured on our machine:
- `IIPFEQ` (contains P/F): 7 NaN columns hit (indices 1604, 1605, 1608, 1609, 1611, 1655, 1656)
- `NPQGGR` (contains G): 1 NaN column hit (index 2784)

### Suggested Fix

Add NaN handling in `collate_fn` of `model/APNet/data_module.py`:

```python
batch_aaindex = self.tokenizer.aaindex.batch_encode(batch_seqs)[:, self.selected_aaindex]
batch_aaindex = torch.from_numpy(batch_aaindex).float()
batch_aaindex = torch.nan_to_num(batch_aaindex, nan=0.0)   # added
```

After the fix: `IIPFEQ` → 0.0028, `NPQGGR` → 0.0001 (both non-amyloid), and full Hex142 metrics recover to ACC 0.852 / AUC 0.909 (paper Table 1: 0.873 / 0.913).
I need to confirm whether the official training pipeline also encounters these NaNs — if so, zero-filling is consistent with training; if not, please share how missing AAindex values were handled during training so we can align the inference pipeline.
